app/internal/import_file.py

```diff
@@ -0,0 +1,185 @@
+                                      = MAX_EVENTS_START_DATE) -> bool:
+    """
+    checking after importing that there is no larger quantity of events
+    with the same date according to the rule we have set.
+    """
+    same_date_counter = 1
+    date_n_count = {}
+    for event in events:
+        if event["S_Date"] in date_n_count:
+            date_n_count[event["S_Date"]] += 1
+            if date_n_count[event["S_Date"]] > same_date_counter:
+                same_date_counter = date_n_count[event["S_Date"]]
+        else:
+            date_n_count[event["S_Date"]] = 1
+    return same_date_counter <= max_event_start_date
+
+
+def import_txt_file(txt_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(txt_file, "r") as text:
+        events = text.readlines()
+    for event in events:
+        if not is_event_text_valid(event):
+            return list()
+        head, content, start_date, end_date = event.split(", ")
+        if not is_date_in_range(start_date) or \
+           not is_date_in_range(end_date.replace("\n", "")):
+            return list()
+        start_date = datetime.datetime.strptime(start_date, "%m-%d-%Y")
+        end_date = datetime.datetime.strptime(end_date.replace("\n", ""),
+                                              "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "S_Date": start_date,
+                                 "E_Date": end_date})
+    return calendar_content
+
+
+def import_ics_file(ics_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(ics_file, "r") as ics:
+        try:
+            calendar_read = Calendar.from_ical(ics.read())
+        except (IndexError, ValueError):
+            return list()
+        for component in calendar_read.walk():
+            if component.name == "VEVENT":
+                if str(component.get('summary')) is None or \
+                   component.get('dtstart') is None or \
+                   component.get('dtend') is None or \
+                   not is_date_in_range(component.get('dtstart').dt) or \
+                   not is_date_in_range(component.get('dtend').dt):
+                    return list()
+                else:
+                    calendar_content.append({
+                        "Head": str(component.get('summary')),
+                        "Content": str(component.get('description')),
+                        "S_Date": component.get('dtstart').dt
+                        .replace(tzinfo=None),
+                        "E_Date": component.get('dtend').dt
+                        .replace(tzinfo=None)
+                    })
+    return calendar_content
+
+
+def save_events_to_database(events: List[Dict[str, Union[str, Any]]],
+                            user_id: int,
+                            session: SessionLocal = SessionLocal()) -> None:
```

Comment:
Prefer not to have mutable variable as default value, it can sometimes act [a bit strange](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/#:~:text=In%20Python%2C%20when%20passing%20a,or%20even%20a%20class%20instance.) in Python.

Reply:
Thank you, I fixed it